template/packages/backend/lib/crypto.ts

```diff
@@ -0,0 +1,68 @@
+import Scrypt from 'scrypt-kdf';
+import nacl from 'tweetnacl';
+
+import * as Time from '@/backend/lib/time';
+
+interface KeyPair {
+  publicKey: Uint8Array;
+  secretKey: Uint8Array;
+}
+
+interface Message {
+  id: string;
+  exp: number;
+}
+
+function bytesOfUTF8(s: string) {
+  return Uint8Array.from(Buffer.from(s, 'utf8'));
+}
+
+function utf8OfBytes(bytes: Uint8Array) {
+  return Buffer.from(bytes).toString('utf8');
+}
+
+function hexOfBytes(bytes: Uint8Array) {
+  return Buffer.from(bytes).toString('hex');
+}
+
+function bytesOfHex(hex: string) {
+  return Uint8Array.from(Buffer.from(hex, 'hex'));
+}
+
+export class CryptoService {
+  private keyPair: KeyPair;
+
+  constructor(private expirationInDays: number) {
+    this.keyPair = nacl.sign.keyPair();
```

Comment:
We'll eventually persist this key-pair to the database, once the backend sprouts database access